android-components/components/concept/toolbar/src/main/java/mozilla/components/concept/toolbar/Toolbar.kt

```diff
@@ -279,26 +279,46 @@ interface Toolbar : ScrollableToolbar {
         private val longClickListener: (() -> Unit)? = null,
         private val listener: () -> Unit,
     ) : Action {
+        private var view: WeakReference<AppCompatImageButton>? = null
+
+        override fun createView(parent: ViewGroup): View =
+            AppCompatImageButton(parent.context).also { imageButton ->
+                view = WeakReference(imageButton)
+
+                imageButton.setImageDrawable(imageDrawable)
+                imageButton.contentDescription = contentDescription
+                imageButton.setTintResource(iconTintColorResource)
+                imageButton.setOnClickListener { listener.invoke() }
+                imageButton.setOnLongClickListener {
+                    longClickListener?.invoke()
+                    true
+                }
+                imageButton.isLongClickable = longClickListener != null
 
-        override fun createView(parent: ViewGroup): View = AppCompatImageButton(parent.context).also { imageButton ->
-            imageButton.setImageDrawable(imageDrawable)
-            imageButton.contentDescription = contentDescription
-            imageButton.setTintResource(iconTintColorResource)
-            imageButton.setOnClickListener { listener.invoke() }
-            imageButton.setOnLongClickListener {
-                longClickListener?.invoke()
-                true
-            }
-            imageButton.isLongClickable = longClickListener != null
+                val backgroundResource = if (background == 0) {
+                    parent.context.theme.resolveAttribute(android.R.attr.selectableItemBackgroundBorderless)
+                } else {
+                    background
+                }
 
-            val backgroundResource = if (background == 0) {
-                parent.context.theme.resolveAttribute(android.R.attr.selectableItemBackgroundBorderless)
-            } else {
-                background
+                imageButton.setBackgroundResource(backgroundResource)
+                padding?.let { imageButton.setPadding(it) }
             }
 
-            imageButton.setBackgroundResource(backgroundResource)
-            padding?.let { imageButton.setPadding(it) }
+        /**
+         * Changes the content description and the tint colour of the view.
+         *
+         * @param contentDescription The content description to use.
+         * @param tintColorResource ID of color resource to tint the icon.
+         */
+        fun updateView(
```

Comment:
Please add tests for this and a changelog entry. We're essentially exposing a function that allows for callback or fine-grain control of updating the view. Normally, we would expect a toggle button to immediate just change colors if it was selected or not. However, I am guessing the reasoning you went with exposing this fine-grain control is to ensure that the translation processing went through and to display an accurate status. Could you confirm this was the intention?

Reply:
I am not quite sure I follow. In [updateView](https://github.com/mozilla-mobile/firefox-android/pull/5600/files#diff-d6281fdb81a5995139750c3bb8316ab4494ebfac844287ff0c691bf805661e81R269), we are changing the tint colours based on the translation state. The toggle buttons would allow you to similarly sent the tint for the selected or unselected state.

Reply:
Yes, but the colour is changing when the user presses the icon. 
    /**
         * Changes the selected state of the action to the inverse of its current state.
         *
         * @param notifyListener If true (default) the listener will be notified about the state change.
         */
        fun toggle(notifyListener: Boolean = true) {
            setSelected(!selected, notifyListener)
        }